dask/bag/core.py

```diff
@@ -244,19 +246,34 @@ def __init__(self, dsk, name, npartitions):
         self.npartitions = npartitions
         self.str = StringAccessor(self)
 
-    def map(self, func):
+    def map(self, func, *args):
         """ Map a function across all elements in collection
 
         >>> import dask.bag as db
         >>> b = db.from_sequence(range(5))
         >>> list(b.map(lambda x: x * 10))  # doctest: +SKIP
         [0, 10, 20, 30, 40]
+
+        Any additional arguments get passed to the function _before_ the data
+        argument; argument values may either by concrete or a dask computation.
```

Comment:
This seems error prone to me, I would have expected it to go the other way.  I'm not suggesting that we change it, just bringing up the point that intuition may be split on this, which is troublesome.

I wonder if a full `db.map` function might be less ambiguous.  Thoughts?